File: Solution with Bitwise Operations.go

package main

var ID_RANGE = [2]int{1, 100}

func maxStudentsOnBench(students [][]int) int {

    maxStudentsOnBench := 0
    frequencyStudentsPerBench := make([]int, ID_RANGE[1]+1)
    bitStampForSeatedStudentsOnBench := make([][]int64, ID_RANGE[1] + 1)
    for i := range bitStampForSeatedStudentsOnBench {
        bitStampForSeatedStudentsOnBench[i] = make([]int64, 2)
    }

    for _, current := range students {

        studentID := current[0]
        benchID := current[1]

        index := studentID / (1 + ID_RANGE[1]/2)
        bitShifts := studentID % (1 + ID_RANGE[1]/2)
        bitStampStudent := int64(1) << bitShifts

        if !isSeated(bitStampForSeatedStudentsOnBench[benchID][index], bitStampStudent) {
            frequencyStudentsPerBench[benchID]++
            bitStampForSeatedStudentsOnBench[benchID][index] ^= bitStampStudent
            maxStudentsOnBench = max(maxStudentsOnBench, frequencyStudentsPerBench[benchID])
        }
    }

    return maxStudentsOnBench
}

func isSeated(bitStampForSeatedStudentsOnBench int64, bitStampStudent int64) bool {
    return (bitStampForSeatedStudentsOnBench & bitStampStudent) != 0
}

File: Solution with Bitwise Operations.js

/**
 * @param {number[][]} students
 * @return {number}
 */
var maxStudentsOnBench = function (students) {
    const ID_RANGE = [1, 100];
    let maxStudentsOnBench = 0;
    const frequencyStudentsPerBench = new Array(ID_RANGE[1] + 1).fill(0);
    const bitStampForSeatedStudentsOnBench = Array.from(new Array(ID_RANGE[1] + 1), () => new Array(4).fill(0));

    for (let [studentID, benchID] of students) {

        const index = Math.floor(studentID / (1 + Math.floor(ID_RANGE[1] / 4)));
        const bitShifts = studentID % (1 + Math.floor(ID_RANGE[1] / 4));
        const bitStampStudent = 1 << bitShifts;

        if (!isSeated(bitStampForSeatedStudentsOnBench[benchID][index], bitStampStudent)) {
            ++frequencyStudentsPerBench[benchID];
            bitStampForSeatedStudentsOnBench[benchID][index] ^= bitStampStudent;
            maxStudentsOnBench = Math.max(maxStudentsOnBench, frequencyStudentsPerBench[benchID]);
        }
    }

    return maxStudentsOnBench;
};

/**
 * @param {number} bitStampForSeatedStudentsOnBench
 * @param {number} bitStampStudent
 * @return {boolean}
 */
function isSeated(bitStampForSeatedStudentsOnBench, bitStampStudent) {
    return (bitStampForSeatedStudentsOnBench & bitStampStudent) !== 0;
}
